Content.Client/Weapons/Melee/MeleeWeaponSystem.cs

```diff
@@ -95,10 +96,29 @@ public override void Update(float frameTime)
         // it's kinda tricky.
         // I think as long as we make secondaries their own component it's probably fine
         // as long as guncomp has an alt-use key then it shouldn't be too much of a PITA to deal with.
-        if (TryComp<GunComponent>(weaponUid, out var gun) && gun.UseKey)
+
+        //Frontier: better support melee vs. ranged checks
+        /*if (TryComp<GunComponent>(weaponUid, out var gun) && gun.UseKey)
         {
             return;
+        }*/
+
+        // Ranged component has priority over melee if both are supported.
+        bool gunBoundToUse = false;
+        bool gunBoundToAlt = false;
```

Comment:
I have a version that uses an enumeration up here: [(f50dadad)](https://github.com/whatston3/frontier-station-14/commit/f50dadadacc5aa538a7254a49a67010b03e0339e)

Either's fine by me, but I do think the comparisons on line 142, 163 are cleaner with two booleans.